lib/access_import/access_import.rb

```diff
@@ -3,41 +3,71 @@ class AccessImport
 
   def initialize(path)
     @acdb = Mdb.open(path)
-    class_accessor(*[Laender, Sprachen, SprachKenntnisse, PersonenRolle, Plz, FamilienRollen,
-                     Begleitete, KostenTraeger, Journale, FallstelleTeilnehmer, Stundenerfassung, Ausbildungen,
-                     FreiwilligenEinsaetze, FreiwilligenEntschaedigung, Kontoangaben].map { |cl| cl.new(@acdb) })
+    class_accessor(
+      *[
+        Laender, Sprachen, SprachKenntnisse, PersonenRolle, Plz, FamilienRollen, Begleitete,
+        KostenTraeger, Journale, FallstelleTeilnehmer, Stundenerfassung, Ausbildungen,
+        FreiwilligenEinsaetze, FreiwilligenEntschaedigung, Kontoangaben
+      ].map { |cl| cl.new(@acdb) }
+    )
     class_accessor(SpracheProHauptperson.new(@acdb, @sprachen, @sprach_kenntnisse),
       EinsatzOrte.new(@acdb, @plz))
     class_accessor(HauptPerson.new(@acdb, @plz, @laender, @sprache_pro_hauptperson))
   end
 
   def make_clients
-    client_transformer = ClientTransform.new(@begleitete, @haupt_person, @familien_rollen)
-    client_count_before = Client.count
-    @personen_rolle.all_clients.each do |key, ac_client|
-      next if Import.where(
-        importable_type: 'Client', access_id: ac_client[:pk_PersonenRolle].to_i
-      ).any?
-      client_attrs = client_transformer.prepare_attributes(ac_client)
-      client = Client.new(client_attrs)
-      client.user = User.first
-      puts "Importing personen_rolle #{key} to Client.id: #{client.id}" if client.save!
+    transformer = ClientTransform.new(@begleitete, @haupt_person, @familien_rollen)
+    make(@personen_rolle.all_clients, transformer, Client, :client_import_handler)
```

Comment:
Yes exactly, instead of `send(import_handler, import_record, entity)` you do `yield import_record, entity` and inline the `#volunteer_import_handler`method as a block.

```ruby
def make_volunteers
  transformer = VolunteerTransform.new(@haupt_person)
  make(@personen_rolle.all_volunteers, transformer, Volunteer) do |volunteer, personen_rolle|
    # ...
  end
end
```

If you would like to keep the `#volunteer_import_handler` method you can also pass the method as a proc argument and call it with `proc.call(..., ...)` instead of yield.